### PR TITLE
Remove stringx package

### LINKED ISSUE #21293: Deprecate and remove stringx package
The stringx package in the Go SDK consists of a handful of unused helper functions that wrap extremely simple functionality. This package should be marked deprecated and removed in a future Beam release.

Imported from Jira [BEAM-13969](https://issues.apache.org/jira/browse/BEAM-13969). Original Jira may contain additional context.
Reported by: jrmccluskey.

### PR DESCRIPTION
**Pull request by damccorm**
We've said this will is deprecated since February. This removes it and replaces one usage in the proper codebase and a couple in examples that had crept in.

Fixes #21293

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by damccorm** [comment]
R: @jrmccluskey

File: sdks/go/examples/native_wordcap/wordcap.go [hunk 1]
```diff
@@ -50,6 +49,11 @@ var (
 	}
 )
 
+// FromBytes converts a byte slice to a string.
+func FromBytes(b []byte) string {
```
Comment (jrmccluskey) [inline sdks/go/examples/native_wordcap/wordcap.go:53]: Is there a reason you fully redefined the function here and then put it in-line in the ParDo?

Comment (damccorm) [reply sdks/go/examples/native_wordcap/wordcap.go]: Nope - bad edit 😳  Good catch, fixed

**Review by jrmccluskey — APPROVED** [review]
LGTM, just waiting on the usual checks to pass before merging

**Pull request merged**